tobac/utils.py

```diff
@@ -556,57 +558,53 @@ def add_coordinates(t, variable_cube):
         elif variable_cube.coord(coord).ndim == 2:
 
             if variable_cube.coord_dims(coord) == (hdim_1, hdim_2):
-                f = interp2d(dimvec_2, dimvec_1, variable_cube.coord(coord).points)
-                coordinate_points = np.asarray(
-                    [f(a, b) for a, b in zip(t["hdim_2"], t["hdim_1"])]
-                )
-
-            if variable_cube.coord_dims(coord) == (hdim_2, hdim_1):
-                f = interp2d(dimvec_1, dimvec_2, variable_cube.coord(coord).points)
-                coordinate_points = np.asarray(
-                    [f(a, b) for a, b in zip(t["hdim_1"], t["hdim_2"])]
-                )
+                points = (dimvec_2, dimvec_1)
+                values = variable_cube.coord(coord).points
+                f = RegularGridInterpolator(points, values)
+                coordinate_points = f((t["hdim_2"], t["hdim_1"]))
 
         # interpolate 3D coordinates:
         # mainly workaround for wrf latitude and longitude (to be fixed in future)
 
         elif variable_cube.coord(coord).ndim == 3:
 
             if variable_cube.coord_dims(coord) == (ndim_time, hdim_1, hdim_2):
-                f = interp2d(
-                    dimvec_2, dimvec_1, variable_cube[0, :, :].coord(coord).points
-                )
-                coordinate_points = [f(a, b) for a, b in zip(t["hdim_2"], t["hdim_1"])]
+                x = dimvec_2
+                y = dimvec_1
+                z = variable_cube[0, :, :].coord(coord).points
+                f = RectBivariateSpline(x, y, z)
+                coordinate_points = [f.ev(a, b) for a, b in zip(t["hdim_2"], t["hdim_1"])]
 
             if variable_cube.coord_dims(coord) == (ndim_time, hdim_2, hdim_1):
                 f = interp2d(
                     dimvec_1, dimvec_2, variable_cube[0, :, :].coord(coord).points
                 )
```

Comment:
**chore**: We should replace this call to `interp2d` with a call to `RegularGridInterpolator`.